packages/patterns/README.md

```diff
@@ -1,11 +1,23 @@
 # `@endo/patterns`
 
 Defines new `Passable` data types and their encoding into the `Tagged` objects defined by the `@endo/pass-style` package. The `@endo/pass-style` package defines the lower level of abstraction on which we need broad agreement for interoperability. The higher level data types defined by this package include
-   - `CopyMap`, `CopySet`, `CopyBag` -- new container types, in addition to the `CopyArrag` and `CopyRecord` already defined by `@endo/pass-style`.
+   - `CopyMap`, `CopySet`, `CopyBag` -- new container types, in addition to the `CopyArray` and `CopyRecord` already defined by `@endo/pass-style`.
    - a variety of Matchers, for expression patterns that can match over Passables
-   - `Key` -- those Passables that can be keys in CopyMaps, CopySets, CopyBags, as well as MapStores and SetStores.
-   - `Pattern` -- includes both literal pass-by-copy structures as well as Matchers. A Pattern *matches* some subset of Passable objects.
+   - `Key` -- Passables that can be keys in CopyMaps, CopySets, CopyBags, as well as MapStores and SetStores.
+   - `Pattern` -- values that *match* some subset of Passables. Includes Matchers along with literal pass-by-copy structures that match theirselves.
 
-The main export from the package is an `M` namespace object, for making a variety of Matchers (hence "M") but also guards, such as the InterfaceGuards used by the `@endo/exo` package to serve as the first level of defense for Exo objects --- those exposed to external messages. The InterfaceGuards use patterns for enforcing type-like restrictions on the arguments and results of these potentially malicious messages.
+The main export from the package is an `M` namespace object, for making a variety of Matchers (hence "M").
 
-See [types.js](./src/types.js) for the definitions of these new types and the methods of the exported `M` namespace object.
+`M` can also make _Guards_ that use Patterns to characterize dynamic behavior such as method argument/response signatures and promise awaiting. The `@endo/exo` package uses InterfaceGuards (each of which maps a collection of method names to their corresponding method guards) as the first level of defense for Exo objects against malformed input. For example:
+```js
+const asyncSerializerI = M.interface('AsyncSerializer', {
+  toString: M.callWhen(M.await(M.any())).returns(M.string()),
+});
+const asyncSerializer = makeExo('AsyncSerializer', asyncSerializerI, {
+  // M.callWhen() delays invocation of this method implementation
+  // while externally-provided input is unsettled.
+  toString(val) { return String(val); },
+});
+```
+
+See [types.js](./src/types.js) for the definitions of these new types and (at typedef `PatternMatchers`) the methods of the exported `M` namespace object.
```

Comment:
Actually, the `PatternMatchers` type only lists the pattern matching methods. The overall type of the `M` object is something like `MatcherNamespace`. You should look.

Reply:
I did look, but forgot the separation by the time I was making edits here. :upside_down_face:

Fixed.